From 9bedad2994a76fc4887d55c7546a3b0c25f841a8 Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Sun, 9 Mar 2014 20:38:03 -0700
Subject: [PATCH 13/13] ARM: Ambarella: Use alternate SPI chip selects

On the Hero3 / Hero3+ Black, the ST7484 LCD uses SPI3
chip select 7, and no other SPI slaves are present (that
we are aware of). Thus, disable all chip selects other than
CS7 to prevent potential GPIO conflicts.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/mach-ambarella/init-filbert.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-ambarella/init-filbert.c b/arch/arm/mach-ambarella/init-filbert.c
index a2261e0..ebe23d1 100644
--- a/arch/arm/mach-ambarella/init-filbert.c
+++ b/arch/arm/mach-ambarella/init-filbert.c
@@ -289,8 +289,14 @@ static void __init ambarella_init_filbert(void)
 	ambarella_platform_sd_controller0.slot[1].gpio_wp.gpio_id = GPIO(76);
 
 	spi2_pdata = (struct ambarella_spi_platform_info *)ambarella_spi2.dev.platform_data;
-	spi2_pdata->cs_num = 7;
-	spi2_pdata->cs_pins[7] = -1;
+
+	/*
+	 * To the best of our knowledge, the only device on SPI3 is the ST7585 LCD.
+	 * It's probably best to leave the remaining chip selects alone.
+	 */
+	for (i = 0; i < 7; i++)
+		spi2_pdata->cs_pins[i] = -1;
+
 	spi_register_board_info(ambarella_spi_devices,
 		ARRAY_SIZE(ambarella_spi_devices));
 
-- 
1.7.9.5

